### PR TITLE
Feature/wil/report gen

### PR DESCRIPTION
**Pull request by lgraa**
# 
## Proposed changes

A lambda function that retrieves information about a batch job from a database, generates an html report about it, saves it to an S3 bucket, and adds the batch job to a queue (so that its files can be later compressed by fileCompress).

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] DevOps related

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [project guidelines](https://github.com/Visual-Testing-for-Android-Apps/Project-guidelines) doc
- [x] I locally linted my code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

Function is set up to be triggered by a SQS queue, although it hasn't yet been tested with an actual queue. Testing was done by evoking the function with test events (similar to ones an SQS queue should send) through the lambda function web page. 
Reports are saved in a subfolder in the bucket, as <batch_job_id>/report.html. If a report.html for a given batch id already exists, calling the function overwrites the file. The function fails if an invalid batch_id is given.
Currently, the function only adds the batch id to a queue; fileCompress isn't set up to be triggered by the queue.

### CODE REVIEW

**Comment by wil-vanderstaay** [comment]
The commented .husky/pre-commit file was pushed to the branch by accident when I could not commit changes due to formatting issues, but I think to merge it should be uncommented. Matti recently removed ts-standard so maybe this doesn't matter now I am not sure. Perhaps someone else can confirm?

**Comment by mt-empty** [comment]
@wil-vanderstaay , I checkout this branch and uncommented .husky/pre-commit, but I didn't encounter any problem. 
Commenting these lines will remove the any precommit hooks, which include `prettier` and `standardjs --fix`.

File: server/postProcessHandler/Makefile [hunk 1]
```diff
@@ -6,9 +6,9 @@ build-PostProcessHandler:
 	npm run fast-compile
 	npm run test
 	npm prune --production
-ifeq ($(OS),Windows_NT)
-	Xcopy /E * $(ARTIFACTS_DIR)
-else
-	cp -r dist "$(ARTIFACTS_DIR)/"
+# ifeq ($(OS),Windows_NT)
```
Comment (reteppeter) [inline server/postProcessHandler/Makefile:9]: You don't have to run this makefile if it is having issues. I wrote an equivalent PowerShell version if you need it.

File: server/uploadHandler/Makefile [hunk 2]
```diff
@@ -6,8 +6,8 @@ build-UploadHandler:
 	npm run fast-compile
 	npm run test
 	npm prune --production
-ifeq ($(OS),Windows_NT)
-	Xcopy /E * $(ARTIFACTS_DIR)
-else
-	cp -R * $(ARTIFACTS_DIR)
-endif
+# ifeq ($(OS),Windows_NT)
```
Comment (reteppeter) [inline server/uploadHandler/Makefile:9]: Same as other instance.

**Review by reteppeter — CHANGES_REQUESTED** [review]

File: server/postProcessHandler/Makefile (hunk 1 above)
Comment (lgraa) [reply server/postProcessHandler/Makefile]: There were issues running it, but they've now been fixed

**Review by reteppeter — APPROVED** [review]

File: server/reportGenerator/src/reportGen.ts [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+		Key: {
+			id: { S: key },
+		},
+	};
+	const dbRes = await getItem(request);
+
+	// construct HTML report contents
+	let res = "<!DOCTYPE html><html lang='en'><head></head><body><div>";
+	if (dbRes.Item != null) {
+		const files = dbRes.Item.files;
```
Comment (beining00) [inline server/reportGenerator/src/reportGen.ts:103]: `item = Converter.unmarshell(dbRes.Item)`. It convert dynamodb json back to javascript json. Then you can just access each property directly (`element.M.fileRef`  becomes `element.fileRef`)

**Review by beining00 — APPROVED** [review]

**Review by CollinsMathews — CHANGES_REQUESTED** [review]
Report that is generated and stored in the s3 bucket is empty? Requesting changes so this isn't merged until it can get solved

**Review by CollinsMathews — APPROVED** [review]
It was on my end, my bad! However, good to know it works on my local machine and AWS console!

**Pull request merged**